iree/compiler/Codegen/SPIRV/SPIRVTile.cpp

```diff
@@ -147,16 +332,32 @@ class SPIRVTilePass final : public SPIRVTileBase<SPIRVTilePass> {
       });
     }
 
+    {
+      RewritePatternSet patterns(context);
+      patterns.insert<linalg::ExtractSliceOfPadTensorSwapPattern>(
+          context, [](tensor::ExtractSliceOp) { return false; });
+      (void)applyPatternsAndFoldGreedily(funcOp, std::move(patterns));
+
+      LLVM_DEBUG({
+        llvm::dbgs() << "--- After fusing padding into consumers ---\n";
+        funcOp.print(llvm::dbgs(), OpPrintingFlags().useLocalScope());
+        llvm::dbgs() << "\n\n";
+      });
+    }
+
     {
       RewritePatternSet patterns =
           linalg::getLinalgTilingCanonicalizationPatterns(context);
       // Pulling in upstream scf.for and affine.min canonicalization patterns.
       // They work on tiled (but not distributed) loops. We only tiled reduction
       // loops previously so this should be fine.
       scf::populateSCFForLoopCanonicalizationPatterns(patterns);
-      if (failed(applyPatternsAndFoldGreedily(funcOp, std::move(patterns)))) {
-        return signalPassFailure();
-      }
+      // Pulling in flow.dispatch.tensor.load/store op canonicalization
+      // patterns. Tiling can generate dim ops taking them as operands.
+      IREE::Flow::populateFlowDispatchCanonicalizationPatterns(patterns,
```

Comment:
I dont think we should pull in all of these. These include some closure optimization patterns that are dangerous. Maybe split the patterns and pull in what you need?

Reply:
This only pulls in `DispatchTensorLoadOp` canonicalization patterns. I checked that it only contains lightweight shape/dim management patterns. Not sure there are closure patterns. 

https://github.com/google/iree/blob/5e2d790/iree/compiler/Dialect/Flow/IR/FlowOpFolders.cpp#L220-L227

Changed to directly pull in `DipsatchTensorLoadOp`'s canonicalization patterns to be clear.